official/vision/beta/projects/yolo/dataloaders/yolo_input.py

```diff
@@ -0,0 +1,511 @@
+  num_detections = tf.shape(classif)[0]
+  return classif, box, areas, iscrowds, num_detections
+
+
+class Parser(parser.Parser):
+  """Parse the dataset in to the YOLO model format. """
+
+  def __init__(
+      self,
+      output_size,
+      masks,
+      anchors,
+      strides,
+      anchor_free_limits=None,
+      max_num_instances=200,
+      area_thresh=0.1,
+      aug_rand_hue=1.0,
+      aug_rand_saturation=1.0,
+      aug_rand_brightness=1.0,
+      letter_box=False,
+      random_pad=True,
+      random_flip=True,
+      jitter=0.0,
+      aug_scale_min=1.0,
+      aug_scale_max=1.0,
+      aug_rand_translate=0.0,
+      aug_rand_perspective=0.0,
+      aug_rand_angle=0.0,
+      anchor_t=4.0,
+      scale_xy=None,
+      best_match_only=False,
+      coco91to80=False,
+      darknet=False,
+      use_tie_breaker=True,
+      dtype='float32',
+      seed=None,
+  ):
+    """Initializes parameters for parsing annotations in the dataset.
+
+    Args:
+      output_size: `Tensor` or `List` for [height, width] of output image. The
+        output_size should be divided by the largest feature stride 2^max_level.
+      masks: `Dict[List[int]]` of values indicating the indexes in the 
+        list of anchor boxes to use an each prediction level between min_level 
+        and max_level. each level must have a list of indexes.  
+      anchors: `List[List[Union[int, float]]]` values for each anchor box.
+      strides: `Dict[int]` for how much the model scales down the images at the
+        largest level.
+      anchor_free_limits: `List` the box sizes that will be allowed at each FPN 
+        level as is done in the FCOS and YOLOX paper for anchor free box 
+        assignment. Anchor free will perform worse than Anchor based, but only 
+        slightly.
+      max_num_instances: `int` for the number of boxes to compute loss on.
+      area_thresh: `float` for the minimum area of a box to allow to pass 
+        through for optimization.
+      aug_rand_hue: `float` indicating the maximum scaling value for 
+        hue. saturation will be scaled between 1 - value and 1 + value.
+      aug_rand_saturation: `float` indicating the maximum scaling value for 
+        saturation. saturation will be scaled between 1/value and value.
+      aug_rand_brightness: `float` indicating the maximum scaling value for 
+        brightness. brightness will be scaled between 1/value and value.
+      letter_box: `boolean` indicating whether upon start of the datapipeline 
+        regardless of the preprocessing ops that are used, the aspect ratio of 
+        the images should be preserved.  
+      random_pad: `bool` indiccating wether to use padding to apply random 
+        translation true for darknet yolo false for scaled yolo.
+      random_flip: `boolean` indicating whether or not to randomly flip the 
+        image horizontally. 
+      jitter: `float` for the maximum change in aspect ratio expected in 
```

Comment:
Does that apply on on aspect ratio? can you use aspect_ratio_jitter instead if that's the case?